### PR TITLE
automation coverage for SAT-28695

### PR DESCRIPTION
**Pull request by vijaysawant**
### Problem Statement
`$ insights-client --ansible-host=foo.example.com`  fails when execute
BZ - SAT-28695

### Solution
This PR will test this BZ scenario in 6.16.z and 6.15.z

### Related Issues
https://issues.redhat.com/browse/SAT-28695

### PRT test Cases example
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_rhcloud_inventory.py -k 'test_positive_register_insights_client_host'
```

### CODE REVIEW

**Comment by vijaysawant** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_rhcloud_inventory.py -k 'test_positive_register_insights_client_host'
```

File: tests/foreman/cli/test_rhcloud_inventory.py [hunk 1]
```diff
@@ -371,3 +371,36 @@ def test_positive_generate_all_reports_job(target_sat):
         assert task_output[0].result == "success"
     finally:
         target_sat.update_setting('allow_auto_inventory_upload', True)
+
+
+@pytest.mark.rhel_ver_list([7, 8, 9])
```
Comment (vsedmik) [inline tests/foreman/cli/test_rhcloud_inventory.py:376]: We could use @damoore044's floating selection here to pick latest 3 RHELs.
```suggestion
@pytest.mark.rhel_ver_match('N-2')
```

File: tests/foreman/cli/test_rhcloud_inventory.py [hunk 2]
```diff
@@ -371,3 +371,36 @@ def test_positive_generate_all_reports_job(target_sat):
...
+
+    :CaseAutomation: Automated
+
+    """
+    # Test connection of insights client
+    assert rhel_insights_vm.execute('insights-client --test-connection').status == 0
+
+    # Execute insight client command
+    output = rhel_insights_vm.execute(f'insights-client --ansible-host={rhel_insights_vm.hostname}')
+    assert 'Ansible hostname updated' in output.stdout
```
Comment (vsedmik) [inline tests/foreman/cli/test_rhcloud_inventory.py:406]: Can we expect the `0` exit code on success from this command?
```suggestion
    assert output.status == 0
    assert 'Ansible hostname updated' in output.stdout
```

**Review by vsedmik — COMMENTED** [review]
Looks good, just 2 minor things.

**Review by ColeHiggins2 — APPROVED** [review]
Ack pending comments

**Comment by vijaysawant** [comment]
@ColeHiggins2 would like to know your thoughts about this test scenario and test module (`cli/test_rhcloud_inventory.py`) in which this test case has been added.
Is it correct test module to add insights client test scenario? Or should be good to create new file as like UI tests has?

```
CLI Tests:
cli/test_rhcloud_inventory.py

UI Tests:
robottelo/tests/foreman/ui/test_rhc.py
robottelo/tests/foreman/ui/test_rhcloud_insights.py
robottelo/tests/foreman/ui/test_rhcloud_inventory.py
```

Are we planning to add more scenarios in future for RH-Clould?

**Comment by vijaysawant** [comment]
```
trigger: test-robottelo
pytest: tests/foreman/cli/test_rhcloud_inventory.py -k 'test_positive_register_insights_client_host'
```

**Review by LadislavVasina1 — APPROVED** [review]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9507
Build Status: UNSTABLE
PRT Comment: pytest tests/foreman/cli/test_rhcloud_inventory.py -k test_positive_register_insights_client_host --external-logging
Test Result : ===== 2 passed, 32 deselected, 132 warnings, 1 error in 1776.46s (0:29:36) =====
```

**Comment by vsedmik** [comment]
Hmm, looks like we need to add the RHEL10 repo into settings and bellow.
```
09:10:43          distro_repo_map = {
09:10:43              'rhel6': settings.repos.rhel6_os,
09:10:43              'rhel7': settings.repos.rhel7_os,
09:10:43              'rhel8': settings.repos.rhel8_os,
09:10:43              'rhel9': settings.repos.rhel9_os,
09:10:43          }
09:10:43          rhel_repo = distro_repo_map.get(rhel_distro)
09:10:43      
09:10:43          if rhel_repo is None:
09:10:43  >           raise ContentHostError(f'Missing RHEL repository configuration for {rhel_distro}.')
09:10:43  E           robottelo.hosts.ContentHostError: Missing RHEL repository configuration for rhel10.
```
Are you one settings part already @damoore044 ?

**Review by chris1984 — APPROVED** [review]

**Comment by damoore044** [comment]
trigger: test-robottelo
pytest: tests/foreman/cli/test_rhcloud_inventory.py -k 'test_positive_register_insights_client_host'

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9532
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_rhcloud_inventory.py -k test_positive_register_insights_client_host --external-logging
Test Result : ========= 3 passed, 32 deselected, 134 warnings in 2209.39s (0:36:49) ==========
```

**Review by damoore044 — APPROVED** [review]

**Comment by Satellite-QE** [comment]
**PRT Result**
```
Build Number: 9535
Build Status: SUCCESS
PRT Comment: pytest tests/foreman/cli/test_rhcloud_inventory.py -k test_positive_register_insights_client_host --external-logging
Test Result : ========= 3 passed, 32 deselected, 134 warnings in 2281.73s (0:38:01) ==========
```

**Review by vsedmik — APPROVED** [review]

**Pull request merged**